src/mono/mono/metadata/assembly.c

```diff
@@ -2139,7 +2139,7 @@ mono_assembly_request_load_from (MonoImage *image, const char *fname,
 		gchar *tmp_fn;
 
 		tmp_fn = g_strdup (fname);
-		for (size_t i = strlen (tmp_fn) - 1; i >= 0; i--) {
+		for (int i = ((int)strlen (tmp_fn) - 1); i >= 0; i--) {
 			if (tmp_fn [i] == '/')
 				tmp_fn [i] = '\\';
 		}
```

Comment:
```suggestion
	g_strdelimit(tmp_fn, '/', '\\');
```

Reply:
Jupp, took the opportunity to move away from the complete construction. Found one more place with similar logic that could be replaced as well.